### PR TITLE
RELATED: RAIL-4438 add dashboard toolbar

### PR DESCRIPTION
**Pull request by no23reason**
* Add a toolbar component that can hold buttons for manipulation of the plugins when in edit mode
* Extend and rename `useDashboardLoaderWithReload` with options to change plugin loading mode and setting extraPlugins without losing the WIP state of the dashboard
* Use the new toolbar in the dashboard plugin template replacing the custom toolbar there

---

Supported PR commands:

| Command                  | Description             |
| ------------------------ | ----------------------- |
| `ok to test`             | Re-run standard checks  |
| `extended test`          | BackstopJS tests        |
| `extended check sonar`   | SonarQube tests         |
| `extended check cypress` | Cypress E2E tests       |
| `extended check plugins` | Dashboard plugins tests |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/10433/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/12435/ : FAILURE in 9m 46s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/10434/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/12438/ : FAILURE in 14m 30s

**Comment by gdgate** [comment]
Merge Failed.

Zuul merger could not merge this change into the base branch. This is most likely caused by merge conflicts. Please rebase the change and upload the rebased version. In case of further errors, contact the zuul administrator.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/10441/ : SUCCESS in 9s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/12444/ : SUCCESS in 15m 34s

**Comment by gdgate** [comment]
Github ref error: SHA of pull request head checked out from remote PR head ref is different from head commit SHA from Github webhook event.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/10487/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/12495/ : FAILURE in 15m 00s

**Comment by no23reason** [comment]
ok to test

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/10488/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/12496/ : SUCCESS in 13m 57s

**Comment by no23reason** [comment]
ok to test

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/10489/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/12497/ : SUCCESS in 13m 48s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/10491/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/12501/ : SUCCESS in 12m 00s

File: tools/dashboard-plugin-template/src/harness/PluginLoader.tsx [hunk 1]
```diff
@@ -1,23 +1,64 @@
...
+            setExtraPlugins(Plugins);
+        }
+    }, [isPluginEnabled, setExtraPlugins]);
+
+    const { status, error, result } = loaderStatus;
 
-const PluginLoaderBody: React.FC<IPluginLoaderBodyProps> = (props) => {
-    const { status, error, result } = props.loaderStatus;
+    const ToolbarComponent = useMemo<CustomToolbarComponent>(() => {
+        return function CustomToolbar() {
```
Comment (ivanmjartan) [inline tools/dashboard-plugin-template/src/harness/PluginLoader.tsx:42]: It will separate it from memo into component.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

File: tools/dashboard-plugin-template/src/harness/PluginLoader.tsx (hunk 1 above)
Comment (no23reason) [reply tools/dashboard-plugin-template/src/harness/PluginLoader.tsx:37]: We cannot completely avoid this, but I separated the body in a separate component.

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR3172

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul/62/ : FAILURE in 2m 33s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/3328/ : SUCCESS in 20s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/2176/ : SUCCESS in 16m 41s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/12509/ : SUCCESS in 15m 14s

**Pull request merged**